### PR TITLE
Updates for 0.12

### PR DESCRIPTION
**Pull request by Lupino**
This pull request only contains the update for 0.12 @nkly 

#87

### CODE REVIEW

**Comment by akheron** [comment]
Why did you switch to use argonaut instead of `Foreign`?

**Comment by Lupino** [comment]
@akheron when is use `purescript-express` in `0.12`, `purescript-foreign-generic` not support `0.12`

but i found `argonaut` is support `0.12` on that time。

on the other way 
   * `Foreign` contains `javascript class` or other 
   * `Json` is just the raw `json data`

**Comment by nkly** [comment]
@Lupino Thank you for splitting the PR. 

What version of `purescript-foreign-generic` did you try? It seems to me that `v7.0.0.` works with minor updates compared to `purescript-argonaut`. Also, though JS objects are easily encoded/decoded to/from JSON, express.js doesn't state explicitly that `req.body` value is a valid JSON object. In that case I think it's better to use an FFI tool rather then JSON parser to interact with JS objects.

I've made a [PR that puts back `ps-foreign` and `ps-foreign-generic`](https://github.com/Lupino/purescript-express/pull/2); could you please check it and verify that it works for you?

Otherwise I don't have any other objections to your changes. Great job!

File: src/Node/Express/Request.purs [hunk 1]
```diff
@@ -1,249 +1,235 @@
...
 -- | If there are multiple params having equal keys
 -- | return the first one.
-getQueryParam :: forall e. String -> HandlerM (express :: EXPRESS | e) (Maybe String)
+getQueryParam :: String -> HandlerM (Maybe String)
 getQueryParam name = HandlerM \req _ _ -> do
-    params <- liftEff $ queryParams req
-    pure $ getOne params name
-
--- | Get all params from query string having specified key.
-getQueryParams :: forall e. String -> HandlerM (express :: EXPRESS | e) (Array String)
```
Comment (nkly) [inline src/Node/Express/Request.purs]: Could you re-implement this function as well? It is a part of public API and its removal might break somebody's setup.

File: test/Test/Handler.purs [hunk 2]
```diff
@@ -65,6 +65,10 @@ testParams = do
         setupMockApp $ use paramsHandler
         sendTestRequest withoutParams assertTestHeaderAbsent
         sendRequest GET urlWithQueryParam id assertTestHeaderExists
+    testExpress "getQueryParams" $ do
+        setupMockApp $ use paramsHandler
+        sendTestRequest withoutParams assertTestHeaderAbsent
+        -- sendRequest GET urlWithQueryParams id assertTestHeaderExists
```
Comment (Lupino) [inline test/Test/Handler.purs]: I don't how to implement the test on `sendRequest GET urlWithQueryParams id assertTestHeaderExists`

Comment (nkly) [reply test/Test/Handler.purs]: No worries, I can write it later.

**Pull request merged**